pkg/anchor/writer/writer.go

```diff
@@ -611,10 +617,29 @@ func (c *Writer) postOfferActivity(anchorEvent *vocab.AnchorEventType, batchWitn
 
 	postID, err := c.Outbox.Post(offer)
 	if err != nil {
-		return fmt.Errorf("failed to post offer for anchor event[%s]: %w", anchorEvent.Index(), err)
+		return fmt.Errorf("failed to post offer for anchor index[%s]: %w", anchorEvent.Index(), err)
 	}
 
-	logger.Debugf("created pre-announce activity for anchor event[%s], post id[%s]", anchorEvent.Index(), postID)
+	logger.Debugf("created pre-announce activity for anchor index[%s], post id[%s]", anchorEvent.Index(), postID)
+
+	if len(witnessesIRI) == 1 {
+		// The Offer was posted only to the public IRI. This means that it will be persisted
+		// in the ActivityPub Outbox (to be viewed by anyone) but won't be sent to any service.
+		// In this case we can handle the anchor event immediately.
+		logger.Debugf("According to witness policy, no witnesses are required for anchor index[%s]. "+
```

Comment:
Is this is due to *both* the witness policy and having no witnesses in the witness collection?

If there is a witness in the witness collection, we should still try to send the Offer. I assume this is still the case in this PR.

Reply:
No, if the witness policy says no witnesses required then an offer is not sent. I don't know what to do in this case, i.e. what if we have 20 witnesses - do we send to all? Currently we determine (based on policy) the minimal set of witnesses to which to send. Also, if we still send the offer to witnesses, how do we know when to anchor the batch? Currently we anchor immediately after satisfying the policy.

If someone configured their policy to not require witnesses, doesn't that mean they don't want the anchors to be witnessed?